common/src/main/java/com/wynntils/wc/objects/item/ItemIdentificationContainer.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.objects.item;
+
+import com.wynntils.core.webapi.profiles.item.IdentificationContainer;
+import com.wynntils.core.webapi.profiles.item.IdentificationModifier;
+import com.wynntils.core.webapi.profiles.item.ItemProfile;
+import com.wynntils.features.ItemStatInfoFeature;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.wc.objects.SpellType;
+import com.wynntils.wc.utils.IdentificationOrderer;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.*;
+
+public class ItemIdentificationContainer {
+    private static final Pattern ITEM_IDENTIFICATION_PATTERN =
+            Pattern.compile("(^\\+?(?<Value>-?\\d+)(?: to \\+?(?<UpperValue>-?\\d+))?(?<Suffix>%|/\\ds|"
+                    + " tier)?(?<Stars>\\*{0,3}) (?<ID>[a-zA-Z 0-9]+))");
+
+    private ItemProfile item;
+    private IdentificationContainer identification;
+    private IdentificationModifier modifier;
+
+    private String shortIdName;
+    private int value;
+    private boolean isRaw;
+    private int stars;
+    private float percent;
+
+    private Component rawLoreLine;
+    private Component percentLoreLine;
+    private Component rangeLoreLine;
+    private Component rerollLoreLine;
+
+    public ItemIdentificationContainer(
+            ItemProfile item,
+            IdentificationContainer identification,
```

Comment:
The names here are a bit confusing. How does an `ItemIdentificationContainer` relate to an `IdentificationContainer` or an `ItemProfile`? Do we need all of those? Can we find a more consistent naming scheme? (Or maybe we have that already, it's just that I'm too stupid to realize how it's supposed to work?)

Reply:
this name is definitely not super clear - i was struggling to think of something descriptive that fit with our already existing objects.

`ItemIdentificationContainer` represents a specific instance of an ID on an item, including its actual value and formatted representation of that value. `IdentificationContainer` represents the base info of an ID on an item; its value range, what type of ID it is, etc.

put in another way, `ItemIdentificationContainer` is to `WynnGearStack` as `IdentificationContainer` is to `ItemProfile`

Reply:
(The naming discussion here is not really helped by the confused terminology used in Wynncraft and the Wynncraft community, where "ID" is a very overloaded word, meaning a lot of things in differrent contexts.)

Reply:
`IdentificationContainer` is the general info of a specific stat on an item, so `ItemProfile` contains multiple of them, one for each of its stats

as it stands right now, `ItemProfile` does roughly contain 1 and 2, and `WynnGearStack` contains all of 1, 2, and 3.